File: src/main/frontend/src/app/models/deployment-info.ts
import {UserConnector} from './user-connector';
import {RepositoryInfo} from './repository-info';
import {Workflow} from './workflow.model';

export interface DeploymentInfo {
  id?: number;
  deploymentName?: string;
  selectedDeploymentSteps?: string[];
  ghUser?: UserConnector;
  repository?: RepositoryInfo;
  workflow?: Workflow;
  state?: string;
  errors?: string[];
}

File: git-service/src/main/java/at/raphael/entity/DeploymentInfo.java
package at.raphael.entity;

import java.util.List;

public class DeploymentInfo{

    public String deploymentName;

    /**
     * Save all Steps which should be done in the deployment
     * User can decide which steps the program should do
     *
     * "file-path-localization",
     * "dockerfile-creation",
     * "workflow-creation",
     * "docker-compose-creation",
     * "github-push",
     * "server-installation"
     */
    public List<String> selectedDeploymentSteps;

    public UserConnector ghUser;

    public RepositoryInfo repository;

    public Workflow workflow;

    public String state;

    public List<String> errors;


    //region Constructor
    public DeploymentInfo() {
    }
    //endregion

}
